modules/crypt.cpp

```diff
@@ -45,12 +45,24 @@
 #include <znc/IRCNetwork.h>
 
 #define REQUIRESSL 1
-#define NICK_PREFIX_KEY "[nick-prefix]"
+#define NICK_PREFIX_KEY "@nick-prefix@"
 
 class CCryptMod : public CModule {
     CString NickPrefix() {
         MCString::iterator it = FindNV(NICK_PREFIX_KEY);
-        return it != EndNV() ? it->second : "*";
+        /*
+         * Check for different Prefixes to not confuse modules with nicknames
+         * Also check for overlap for rare cases like:
+         * SP = "*"; NP = "*s"; "tatus" sends an encrypted message appearing at "*status"
+         */
+        CString sStatusPrefix = GetUser()->GetStatusPrefix();
+        if (it != EndNV()) {
+            size_t sp = sStatusPrefix.size();
+            size_t np = it->second.size();
+            if (sStatusPrefix.CaseCmp(it->second, (np > sp) ? sp : np ))
```

Comment:
Ah, you're right, I didn't notice. Then let's make it more explicit with `!= 0`.